pkg/controller/render/render_controller.go

```diff
@@ -434,13 +440,20 @@ func (ctrl *Controller) garbageCollectRenderedConfigs(pool *mcfgv1.MachineConfig
 	return nil
 }
 
-
 func (ctrl *Controller) syncGeneratedMachineConfig(pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.MachineConfig) error {
 	if len(configs) == 0 {
 		return nil
 	}
 
-	generated, err := generateMachineConfig(pool, configs)
+	cc, err := ctrl.ccLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("could not enumerate ControllerConfig %s", err)
+	}
+	if len(cc) == 0 {
+		return fmt.Errorf("ControllerConfigList is empty")
+	}
+
+	generated, err := generateMachineConfig(pool, configs, cc[0])
```

Comment:
maybe someone can shed some light on this snipped I found in the kubeconfig/containerruntimeconfig controllers. Is it correct to grab the controllerconfig like this? isn't there always just 1 controller config at any point in time?

Reply:
@cgwalters @kikisdeliveryservice ^^^

Reply:
Yeah, AFAIK there's always just one.  It's basically a way for the template controller to react to just one aggregation of data AIUI.

But I don't see the problem with having the render controller grab it directly like this either.